packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -713,7 +713,7 @@ class CachedArtifacts implements Artifacts {
         if (mode == BuildMode.debug || mode == null) {
           return _fileSystem.path.join(engineDir, platformName);
         }
-        final String suffix = mode != BuildMode.debug ? '-${snakeCase(getModeName(mode), '-')}' : '';
+        final String suffix = mode != BuildMode.debug ? '-${snakeCase(mode.name, '-')}' : '';
```

Comment:
Isn't this `BuildMode` a very good candidate to be Enum-ified? [Source](https://github.com/flutter/flutter/blob/ba8d100597a4b85bf1069da0120ec40d7fa0aae8/packages/flutter_tools/lib/src/build_info.dart#L380).

In all the code this PR is touching, it's being used as an "enum-like" thing :/